src/registration/popupTristate.ts

```diff
@@ -32,28 +29,21 @@ export class PopupTristate implements Disposable {
 
     if (!statePoller.state.currentState) {
       const disposable = statePoller.event((state) => {
-        if (state.currentState) {
+        if (state.currentState?.is_logged_in === false) {
           disposable.dispose();
-          this.isLoggedIn = state.currentState.is_logged_in;
+          this.isLoggedIn = false;
           this.changedStateEmitter.fire();
         }
       });
     }
   }
 
   get needWait(): boolean {
-    return (
-      this.installationState === InstallationState.Undefined ||
-      !statePoller.state.currentState
-    );
+    return this.newInstall === undefined || this.isLoggedIn === undefined;
```

Comment:
What is the effect of this? I'm missing context here.

Reply:
I suggest you read the code :)

Reply:
the capabilities arrive late

the state is determined by events that occur asynchronously. This code waits for the collection of events which is to trigger the relevant notifications

Reply:
👍